### PR TITLE
chore: make adjustments to macos setup script and build for android API v32

### PR DESCRIPTION
**Pull request by daniellacosse**
Still needs a test. I got the Pixel 6 emulator to come up, but the screen was black.

### CODE REVIEW

File: tools/build/setup_macos_android.sh [hunk 1]
```diff
@@ -56,12 +61,13 @@ function install_gradle() {
   if which -s gradle; then
     echo 'Gradle already installed'
   else
-    if ! which -s brew; then
-      # TODO(fortuna): Install gradle without Homebrew: https://gradle.org/install/#manually
-      echo 'Homebrew is not installed. You need to manually install it: https://docs.brew.sh/Installation' > 2
-      exit 1
-    fi
-    brew install gradle
+    declare -r gradle_home=${1?Need to pass Gradle home}
```
Comment (fortuna) [inline tools/build/setup_macos_android.sh:64]: I recommend processing arguments as the first thing in a function, so it's clear.

File: tools/build/setup_macos_android.sh [hunk 2]
```diff
@@ -81,12 +87,13 @@ function main() {
   declare -r android_home="${ANDROID_HOME:-$HOME/Library/Android/sdk}"
   install_android_tools "${android_home}"
 
-  install_gradle
-  gradle --version
+  declare -r gradle_home="${HOME}/opt/gradle"
```
Comment (fortuna) [inline tools/build/setup_macos_android.sh:90]: Is this standard? I've never seen this. Is another directory more appropriate, like $HOME/Library?

File: tools/build/setup_macos_android.sh [hunk 3]
```diff
@@ -81,12 +87,13 @@ function main() {
...
+  install_gradle "${gradle_home}"
+  "${gradle_home}/gradle-7.6/bin/gradle" --version
 
-  echo 'Setup done. Make sure to define these environment variables:'
+  echo 'Setup done. Make these environment variables are defined:'
   echo "export ANDROID_SDK_ROOT=${android_home}"
-  echo 'export PATH="$PATH:${ANDROID_SDK_ROOT}/platform-tools:${ANDROID_SDK_ROOT}/cmdline-tools/8.0/bin:${ANDROID_SDK_ROOT}/emulator"'
+  echo 'export PATH="$PATH:${ANDROID_SDK_ROOT}/platform-tools:${ANDROID_SDK_ROOT}/cmdline-tools/8.0/bin:${ANDROID_SDK_ROOT}/emulator:${HOME}/opt/gradle/gradle-7.6/bin"'
```
Comment (fortuna) [inline tools/build/setup_macos_android.sh:96]: You can refer to gradle_home here

File: tools/build/setup_macos_android.sh [hunk 4]
```diff
@@ -20,22 +20,26 @@ source "$(dirname "$0")/android_tools_versions.sh" || exit
...
   # Cordova Android 10 has to use JDK 11.
-  if [[ "$(javac --version) 2> /dev/null" =~ "javac 11.*" ]]; then
-    echo 'JDK already installed';
+  if [[ "$(javac --version) 2> /dev/null" =~ "javac 11." ]]; then
+    echo 'JDK already installed'
+    return
   fi
 
-  if [[ -d "$HOME/Library/Java/JavaVirtualMachines/jdk-11.0.2.jdk" ]]; then
-    echo 'JDK already installed, but not configured properly. Make sure to set JAVA_HOME.'
```
[the remark this answers is not in the document]
Comment (fortuna) [reply tools/build/setup_macos_android.sh:28]: You are specifying the wrong directory. It should be `$HOME/Library/Java/JavaVirtualMachines/jdk-11.0.2.jdk/Contents/Home`

I'd restore the JAVA_HOME comment.

File: tools/build/setup_macos_android.sh [hunk 5]
```diff
@@ -44,6 +48,7 @@ function install_android_tools() {
     declare -r tmp_zip_dir="$(mktemp -d)"
     curl "https://dl.google.com/android/repository/commandlinetools-mac-9123335_latest.zip" --create-dirs --output "${tmp_zip_dir}/tools.zip"
```
[the remark this answers is not in the document]
Comment (fortuna) [reply tools/build/setup_macos_android.sh:49]: https://developer.android.com/studio#command-line-tools-only

Perhaps we should add a comment mentioning where this came from

File: tools/build/setup_macos_android.sh [hunk 6]
```diff
@@ -53,19 +58,20 @@ function install_android_tools() {
 }
 
 function install_gradle() {
+  declare -r gradle_home=${1?Need to pass Gradle home}
   if which -s gradle; then
     echo 'Gradle already installed'
```
Comment (fortuna) [inline tools/build/setup_macos_android.sh:63]: Return early to avoid nesting.

File: tools/build/setup_macos_android.sh [hunk 7]
```diff
@@ -53,19 +58,20 @@ function install_android_tools() {
...
   else
-    if ! which -s brew; then
-      # TODO(fortuna): Install gradle without Homebrew: https://gradle.org/install/#manually
-      echo 'Homebrew is not installed. You need to manually install it: https://docs.brew.sh/Installation' > 2
-      exit 1
-    fi
-    brew install gradle
+    declare -r tmp_zip_dir="$(mktemp -d)"
```
Comment (fortuna) [inline tools/build/setup_macos_android.sh:65]: Perhaps mention that you are following https://gradle.org/install/#manually

File: tools/build/setup_macos_android.sh (hunk 2 above)
Comment (fortuna) [reply tools/build/setup_macos_android.sh:91]: Yeah,` /opt` is a thing, `$HOME/opt` is not. Lets use `/opt` as the instructions recommend.
```suggestion
  declare -r gradle_home="/opt/gradle"
```

Comment (fortuna) [reply tools/build/setup_macos_android.sh]: Ah, you will need sudo for that. It seems the right place is `/usr/local/opt/gradle`

File: tools/build/setup_macos_android.sh [hunk 8]
```diff
@@ -80,13 +88,17 @@ function main() {
 
   declare -r android_home="${ANDROID_HOME:-$HOME/Library/Android/sdk}"
   install_android_tools "${android_home}"
+  echo
 
-  install_gradle
-  gradle --version
+  # Install location as suggested by https://gradle.org/install/#manually
+  declare -r gradle_home="/opt/gradle"
```
Comment (fortuna) [inline tools/build/setup_macos_android.sh:94]:
```suggestion
  declare -r gradle_home="/usr/local/opt/gradle"
```

**Review by fortuna — APPROVED** [review]

**Review by fortuna — APPROVED** [review]

**Comment by bemasc** [comment]
Once you merge in `master`, I believe you'll need to update src/cordova/android/OutlineAndroidLib/build.gradle and src/cordova/android/OutlineAndroidLib/outline/build.gradle

**Comment by bemasc** [comment]
I believe those files contain version numbers that can now be updated.

**Review by fortuna — APPROVED** [review]

**Pull request merged**